[Bug] [Connector-V2] Updates Iceberg version to 1.6.1 (#9387)

LuigiDurso opened the pull request:
Upgrades the Iceberg dependency to version 1.6.1.

This ensures compatibility with custom oauth token uri.

This pull request updates the Iceberg dependency in the `seatunnel-connectors-v2/connector-iceberg/pom.xml` file to a newer version.

Dependency update:

* Updated the `iceberg.version` property from `1.4.2` to `1.6.1` to ensure compatibility with the latest features and fixes in Iceberg. (`[seatunnel-connectors-v2/connector-iceberg/pom.xmlL33-R33](diffhunk://#diff-cbb0301376499c039b2b41bd654c3a037e2ca60a0b21c0adaabaff7fded4dfceL33-R33)`)

LuigiDurso committed:
[Fix] [Connector-V2] Updates Iceberg version to 1.6.1 (#9387)

Upgrades the Iceberg dependency to version 1.6.1.

This ensures compatibility with custom oauth token uri.

Hisoka-X commented:
Thanks @LuigiDurso ! Could you follow the guide to open ci in your fork repository? https://github.com/apache/seatunnel/pull/9451/checks?check_run_id=44271538658

Hisoka-X commented:
Please re-trigger the failed ci.

Hisoka-X reviewed (commented):
Thanks @LuigiDurso ! Please udpate https://github.com/apache/seatunnel/blob/dev/docs/en/connector-v2/source/Iceberg.md?plain=1#L9 too. And other 3 docs (sink and zh source and sink)

LuigiDurso committed:
Updates Iceberg version support to 1.6.1

Updates the supported Iceberg version in the documentation
for both sink and source connectors in English and Chinese.
